torchtune/dev/rl/workers/datacollectors/sync.py

```diff
@@ -244,16 +250,14 @@ def _log_metrics(
 
         ray.get(self._metric_logger.log_dict.remote(log_dict, step=step_idx))
 
-    def run(self):
+    async def run(self):
         num_steps = (self.cfg.num_steps // self.cfg.vllm.num_workers) + 1
         for i in range(num_steps):
             self.rollout(i)
             if i % self.cfg.vllm.steps_before_sync == 0:
                 log.info(f"{self.worker_id} about to update weights")
-                ray.get(
-                    self.remote_weight_updater.update_weights.remote(
-                        weights=None, worker_ids=self.worker_id
-                    )
+                await self.weight_update_sender.update_weights.remote(
+                    weights=None, worker_ids=self.worker_id
```

Comment:
For some reason had to modify this to prevent hangs when bumping to rl main @vmoens any intuition on why

Reply:
not exactly sure bc it doesn't touch any async code afaict but it looks like a side effect of https://github.com/pytorch/rl/commit/31df775477f5b797ce88b3d07ac2972b6726cc8b